### PR TITLE
Fix CLI spawn tsx ENOENT by scoping source fallback to monorepo

### PR DESCRIPTION
**Pull request by NKoech123**
### Summary
Fixes `npx @agent-native/core create` failing with `spawn tsx ENOENT` by scoping the CLI launcher's source fallback so it only triggers in monorepo source checkouts.

### Problem
When installed via `npx @agent-native/core create`, the CLI launcher's mtime freshness check compared source `.ts` files against compiled `.js` files to decide whether to fall back to running source via `tsx`. Tarball extraction during npm install can leave `.ts` files with newer mtimes than their compiled `.js` counterparts, causing the launcher to incorrectly attempt to spawn `tsx` — a dependency that isn't installed in published packages — resulting in `spawn tsx ENOENT` and a broken CLI.

### Solution
The launcher now only considers the source fallback when running from an actual monorepo source checkout (detected via presence of `tsconfig.cli.json`). Installed packages always run the shipped `dist` build regardless of file mtimes. The fallback decision logic was extracted into a testable pure function (`shouldUseSourceFallback`) in a new `launcher.js` module.

### Key Changes
- Added `packages/core/bin/launcher.js` exporting `shouldUseSourceFallback`, which takes `isSourceCheckout`, `sourceEntryExists`, `distEntryExists`, and precomputed `freshness` data instead of touching the filesystem directly.
- Updated `packages/core/bin/agent-native.js` to detect `isSourceCheckout` via `existsSync(join(binDir, "../tsconfig.cli.json"))` and to only trigger the freshness-based fallback when in a source checkout.
- Added `packages/core/src/cli/launcher.spec.ts` with unit tests covering installed-package regression, missing dist, missing source entry, and freshness comparison edge cases.
- Added a changeset documenting the patch fix.

---

<a href="https://builder.io/app/projects/8a4a0eb9ae3d486cace5/dark-particle-mjix8cma"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cdn.builder.io/api/v1/image/assets%2FYJIGb4i01jvw0SRdL5Bt%2F226fa21c49ce4f95a5aba53aa594fe7a">Edit in Builder</picture></a>&nbsp;&nbsp;<a href="https://8a4a0eb9ae3d486cace5-dark-particle-mjix8cma.projects.builder.my/"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cdn.builder.io/api/v1/image/assets%2FYJIGb4i01jvw0SRdL5Bt%2Fe530b1333b5b4cedac9c41b8573c8268">Preview</picture></a>

---

To clone this PR locally use the [Github CLI](https://cli.github.com/) with command `gh pr checkout 2302`

You can tag me at @builderio for anything you want me to fix or change

### CODE REVIEW

**Review by steve8708 — APPROVED** [review]

File: packages/core/bin/agent-native.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ import { existsSync, statSync } from "node:fs";
 import { dirname, join } from "node:path";
 import { fileURLToPath, pathToFileURL } from "node:url";
 
+import { shouldUseSourceFallback } from "./launcher.js";
```
[the remark this answers is not in the document]
Comment (NKoech123) [reply packages/core/bin/agent-native.js:8]: @builderio-bot verify this issue exists and fix it and let me know the changes you made and why if changes are required. Be skeptical.

File: packages/core/bin/agent-native.js [hunk 2]
```diff
@@ -16,22 +18,39 @@ const freshnessChecks = [
...
+const useSourceFallback = shouldUseSourceFallback({
+  isSourceCheckout,
+  sourceEntryExists: existsSync(sourceEntry),
+  distEntryExists: existsSync(distEntry),
+  freshness,
+});
+
+if (!useSourceFallback && existsSync(distEntry)) {
   await import(pathToFileURL(distEntry).href);
 } else {
```
[the remark this answers is not in the document]
Comment (NKoech123) [reply packages/core/bin/agent-native.js]: @builderio-bot verify this issue and fix it. Be skeptical and let me know what changes you've made if any
